external-table/src/pxfprotocol.c

```diff
@@ -46,6 +48,70 @@ static gphadoop_context *create_context(PG_FUNCTION_ARGS, bool is_import);
 static void cleanup_context(gphadoop_context *context);
 static void check_caller(PG_FUNCTION_ARGS, const char *func_name);
 
+void _PG_init(void);
+static bool check_pxf_base(char **newval, void **extra, GucSource source);
+
+/* GUC variables */
+char *PxfBase = NULL;
+
+/*
+ * _PG_init gets called when the extension is loaded.
+ */
+void
+_PG_init(void)
+{
+	ereport(DEBUG1, (errmsg("PXF extension loaded.")));
+	DefineCustomStringVariable(
+		"pxf.pxf_base",
+		gettext_noop("PXF_BASE directory containing PXF configuration files"),
+		NULL,
+		&PxfBase,
+		"",
+		PGC_SIGHUP,
+		0,
+		check_pxf_base, NULL, NULL);
+}
+
+static bool
+check_pxf_base(char **newval, void **extra, GucSource source)
+{
+	if (newval && strcmp(*newval, "") == 0)
+	{
+		/*
+		 * don't do anything if we have the default value
+		 * if we add something here, there will be an err and the guc value
+		 * will not be set. the check function runs even for the initialization of the default value
+		 * the assumption is that pxf register will take care of setting this guc
+		 */
+		if (source == PGC_S_DEFAULT)
+			return true;
+		else
+		{
+			GUC_check_errcode(ERRCODE_CONFIG_FILE_ERROR);
+			GUC_check_errmsg("invalid value for parameter \"pxf.pxf_base\": parameter cannot be empty");
+			GUC_check_errhint("pxf.pxf_base cannot be empty. Please run `pxf register` to set it.");
```

Comment:
```suggestion
			GUC_check_errhint("Please run \"pxf register\" command to set \"pxf.pxf_base\" parameter.");
```
I don't think we need to repeat "pxf.pxf_base cannot be empty" since this is what the error message says already.